File: Backend/SafalYatra/driver/getDriverProfile.php
<?php
// Including database connection and authentication helper
include "../config/connection.php";
include "../utils/authHelper.php";

// Checking if the token is provided in the request
if (!isset($_POST['token'])) {
    // Sending a failure response if token is missing
    echo json_encode(array(
        "success" => false,
        "message" => "Token is required"
    ));
    die(); // Stopping script execution
}

$token = $_POST['token']; // Retrieving the token value from POST request

$driverId = getDriverId($token); // Getting driver ID using the token (verifying authenticity)

// Checking if a valid driver ID is returned
if (!$driverId) {
    // Sending a failure response if token is invalid or expired
    echo json_encode(array(
        "success" => false,
        "message" => "Invalid token"
    ));
    die(); // Stopping script execution
}


// Preparing a SQL query to fetch driver details 
// and joining car_operators table to get the operator's name
$sql = "SELECT d.driver_name, d.phone_number, d.age, d.gender, d.experience, d.email, d.license_number, d.image_url, d.added_date, o.operator_name
FROM drivers d 
JOIN car_operators o ON d.operator_id = o.operator_id
WHERE d.driver_id = $driverId ";

// Executing the query
$result = mysqli_query($connect, $sql);

// Checking if the query execution was successful
if (!$result) {
    // Sending a failure response if the query failed
    echo json_encode(array(
        "success" => false,
        "message" => "Failed to fetch operators data",

    ));
    die(); // Stopping script execution
}

// Fetching the result as an associative array
$driver = mysqli_fetch_assoc($result);
// Sending a success response along with driver details
echo json_encode(array(
    "success" => true,
    "message" => "Driver data fetched successfully",
    "driver" => $driver

));

File: Frontend/safal_yatra/lib/app/models/getDriverProfile.dart
// To parse this JSON data, do
//
//     final driverProfileResponse = driverProfileResponseFromJson(jsonString);

import 'dart:convert';

DriverProfileResponse driverProfileResponseFromJson(String str) =>
    DriverProfileResponse.fromJson(json.decode(str));

String driverProfileResponseToJson(DriverProfileResponse data) =>
    json.encode(data.toJson());

class DriverProfileResponse {
  final bool? success;
  final String? message;
  final Driver? driver;

  DriverProfileResponse({
    this.success,
    this.message,
    this.driver,
  });

  factory DriverProfileResponse.fromJson(Map<String, dynamic> json) =>
      DriverProfileResponse(
        success: json["success"],
        message: json["message"],
        driver: json["driver"] == null ? null : Driver.fromJson(json["driver"]),
      );

  Map<String, dynamic> toJson() => {
        "success": success,
        "message": message,
        "driver": driver?.toJson(),
      };
}

class Driver {
  final String? driverName;
  final String? phoneNumber;
  final String? age;
  final String? gender;
  final String? experience;
  final String? email;
  final String? licenseNumber;
  final dynamic imageUrl;
  final DateTime? addedDate;
  final String? operatorName;

  Driver({
    this.driverName,
    this.phoneNumber,
    this.age,
    this.gender,
    this.experience,
    this.email,
    this.licenseNumber,
    this.imageUrl,
    this.addedDate,
    this.operatorName,
  });

  factory Driver.fromJson(Map<String, dynamic> json) => Driver(
        driverName: json["driver_name"],
        phoneNumber: json["phone_number"],
        age: json["age"],
        gender: json["gender"],
        experience: json["experience"],
        email: json["email"],
        licenseNumber: json["license_number"],
        imageUrl: json["image_url"],
        addedDate: json["added_date"] == null
            ? null
            : DateTime.parse(json["added_date"]),
        operatorName: json["operator_name"],
      );

  Map<String, dynamic> toJson() => {
        "driver_name": driverName,
        "phone_number": phoneNumber,
        "age": age,
        "gender": gender,
        "experience": experience,
        "email": email,
        "license_number": licenseNumber,
        "image_url": imageUrl,
        "added_date": addedDate?.toIso8601String(),
        "operator_name": operatorName,
      };
}
